<launch>
    <param name="is_respawn" value="false"/>
     
    <arg name="robot_id" default="1"/>
    <!-- <arg name="robot_id" value="$(env ROBOT_ID)"/> -->
    <arg name="ip_to_query" default="192.168.1.11"/>
    <arg name="msg_tp_in" default="submap_raw"/>
    <arg name="msg_tp_out" default="submap"/>
    <arg name="msg_ns" default="submap"/>
    <arg name="logger_level" default="Debug"/>

    <arg name="buffer_size_limit" default="20"/>
    <arg name="timeout_limit_ms" default="5000"/>

    <group ns="robot_$(arg robot_id)">
        <node pkg="buffer_manager" type="buffer_manager_node"  name="buffer_manager" output="screen" respawn="true" ns="$(arg msg_ns)">
            <param name="logger_level" value="$(arg logger_level)"/>
            <param name="msg_ns_" value="$(arg msg_ns)"/>

            <param name="ip_to_query_" value="$(arg ip_to_query)"/>
            <param name="buffer_size_limit_" value="$(arg buffer_size_limit)"/>
            <param name="timeout_limit_ms_" value="$(arg timeout_limit_ms)"/>

            <param name="msg_tp_in_" value="/robot_$(arg robot_id)/$(arg msg_tp_in)"/>
            <param name="msg_tp_out_" value="/robot_$(arg robot_id)/$(arg msg_tp_out)"/>

            <param name="msg_backup_filename_" value="$(find buffer_manager)/data/$(arg msg_ns)_backup.dat"/>
            <param name="last_sent_id_filename_" value="$(find buffer_manager)/data/last_sent_$(arg msg_ns)_id.dat"/>
        </node>
    </group>
</launch>